src/wechaty/plugin.py

```diff
@@ -91,15 +91,29 @@ def __init__(self, options: Optional[WechatyPluginOptions] = None):
             options = WechatyPluginOptions()
         self.options = options
 
+    def set_bot(self, bot: Wechaty):
+        """set bot instance to WechatyPlugin
+
+        Args:
+            bot (Wechaty): the instance of Wechaty
+        """
+        self.bot = bot
+
     async def init_plugin(self, wechaty: Wechaty):
         """set wechaty to the plugin"""
-        self.bot = wechaty
```

Comment:
Since `init_plugin` has no content, why should keep it

Reply:
This is a good question. Why should we need a empty method in `WechatyPlugin` class ? 

> this is the empty method, not a abstract method which must be implemented by derived class. So, derived class can `init plugin` or not, simple application scenario.

The derived instance can call the `init_plugin` method from the base class, which enables all plugins to be initialized uniformly.

check out these code :
https://github.com/wechaty/python-wechaty/blob/af852c3334a340711f2a225933f5d5e3c15dae16/src/wechaty/plugin.py#L302-L311

Reply:
Thanks for your detailed explanation !